frontend/device/devicelistener.lua

```diff
@@ -51,24 +51,38 @@ end
 function DeviceListener:onShowWarmth(value)
     local powerd = Device:getPowerDevice()
     if powerd.fl_warmth ~= nil then
-        UIManager:show(Notification:new{
-            text = T(_("Warmth set to %1."), powerd.fl_warmth),
+        if powerd.fl_warmth_max == nil then
+            UIManager:show(Notification:new{
+                text = T(_("Warmth set to %1."), math.floor(powerd.fl_warmth)),
+                timeout = 1.0,
+            })
+        else
+            UIManager:show(Notification:new{
+            text = T(_("Warmth set to %1."), math.floor(powerd.fl_warmth/100*powerd.fl_warmth_max)),
             timeout = 1.0,
-        })
+            })
+        end
     end
     return true
 end
 
 -- frontlight controller
 if Device:hasFrontlight() then
 
-    -- direction +1 - increase frontlight
-    -- direction -1 - decrease frontlight
-    function DeviceListener:onChangeFlIntensity(ges, direction)
-        local powerd = Device:getPowerDevice()
+    local function calculateGestureDeltaIntervall(ges, direction, min, max)
         local delta_int
-        --received gesture
         if type(ges) == "table" then
+            -- here we are using just two scales
+            -- big scale is for high dynamic ranges (e.g. brightness from 1..100)
+            --           original scale maybe tuned by hand
+            -- small scale is for lower dynamic ranges (e.g. warmth from 1..10)
+            --           scale entries are calculated by math.round(1*sqrt(2)^n)  
+            local steps_fl_big_scale = { 0.1, 0.1, 0.2, 0.4, 0.7, 1.1, 1.6, 2.2, 2.9, 3.7, 4.6, 5.6, 6.7, 7.9, 9.2, 10.6, }
+            local steps_fl_small_scale = { 1.0, 1.0, 2.0, 3.0, 4.0, 6.0, 8.1, 11.3 }
+            local steps_fl = steps_fl_big_scale
+            if ( min - max < 50 ) then
```

Comment:
Unless you write this, those parentheses don't do anything to improve clarity.
```suggestion
            if (min - max) < 50 then
```

Reply:
OK